scripts/solve_network.py

```diff
@@ -854,6 +854,43 @@ def add_existing(n):
             n.generators.loc[tech_index, tech] = existing_res
 
 
+def add_lossy_bidirectional_link_constraints(n: pypsa.components.Network) -> None:

```

Comment:
I did this slightly different (See below). Feel free to check and comment if you do not agree:

```
def add_lossy_bidirectional_link_constraints(n):
    """
        Ensures that the two links simulating a bidirectional_link are extended the same amount.
    """
    if not n.links.p_nom_extendable.any() or "reversed" not in n.links.columns:
        return

    # Ensure 'reversed' column is boolean
    n.links["reversed"] = n.links.reversed.fillna(0).astype(bool)
    carriers = n.links.loc[n.links.reversed, "carrier"].unique()

    # Get forward link indices (non-reversed)
    forward_i = n.links.query(
        "carrier in @carriers and ~reversed and p_nom_extendable"
    ).index

    # Function to get backward (reversed) indices corresponding to forward links
    def get_backward_i(forward_i):
        return pd.Index(
            [
                re.sub(r"-(\d{4})$", r"-reversed-\1", s)
                if re.search(r"-\d{4}$", s)
                else s + "-reversed"
                for s in forward_i
            ]
        )

    backward_i = get_backward_i(forward_i)

    # Get the p_nom optimization variables for the links using the get_var function
    links_p_nom = get_var(n, "Link", "p_nom")

    # Only consider forward and backward links that are present in the optimization variables
    subset_forward = forward_i.intersection(links_p_nom.index)
    subset_backward = backward_i.intersection(links_p_nom.index)

    # Ensure we have a matching number of forward and backward links
    if len(subset_forward) != len(subset_backward):
        raise ValueError("Mismatch between forward and backward links.")

    # For each forward index, find the corresponding backward index and define the constraint
    for fwd, bwd in zip(subset_forward, subset_backward):
        lhs = linexpr((1, links_p_nom.loc[bwd]))  # LHS (backward link)
        rhs = linexpr((-1, links_p_nom.loc[fwd]))  # RHS (forward link with -1 coefficient)

        # Define the constraint for this pair of forward-backward link
        define_constraints(
            n,
            lhs + rhs,  # LHS - RHS = 0 (equality constraint)
            "=",
            0,  # Right-hand side is 0 because we want them to be equal
            name="Link",
            attr=f"bidirectional_sync_{fwd}",  # Custom name for this constraint (per pair)
            axes=[pd.Index([fwd])]  # The axes refer to the forward link
        )
```

Reply:
Hey @Eddy-JV, even though the additional checks you proposed here are not neccessary right now, I think it's great to future-proof the code right now. I'm going to include the additional checks in the final version.
However, I wouldn't add the constraints one by one for each component, since no other constraint is added that way.